lib/app.js

```diff
@@ -173,9 +173,13 @@ if (navigator.mozSetMessageHandler) {
   navigator.mozSetMessageHandler('activity', function(activityRequest) {
     var option = activityRequest.source;
     if (option.name === 'share') {
-      app.activity = option;
       if (initCalled) {
```

Comment:
This seems a bit racy even with the new code. Maybe it would be better to make ```initCalled``` a promise that you call ```then()``` on to handle the activity.